src/librustc/middle/resolve_lifetime.rs

```diff
@@ -2178,7 +2178,6 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
         let arg_lifetimes = inputs
             .iter()
             .enumerate()
-            .skip(has_self as usize)
```

Comment:
Hmm... This breaks the code below:

```
struct Foo;
struct Bar<'a>(&'a Foo);
impl<'a> Bar<'a> {
    fn a(self: Bar<'a>, f: &Foo) -> (Bar<'a>, &Foo) { (self, f) } //~ ERROR E0106
}
```

It should probably skip only if `is_self_ty` is `false`.
https://github.com/rust-lang/rust/blob/589beb979c0f1a9736e02e578913db88398c3ba5/src/librustc/middle/resolve_lifetime.rs#L2159

Reply:
Fixed in the latest commit.